.github/workflows/docs-dev.yaml

```diff
@@ -0,0 +1,65 @@
+name: Deploy the developer documentation
```

Comment:
Would it make sense to put this in the same `docs.yaml` as a new job? 🤷  It looks like a lot of copy & pasting. I don't know if we can share any of it?

Reply:
I guess that could make sense, but I'm not sure how to reuse e.g. the target calculating block between two different jobs anyway — GFYAML doesn't support anchors or merges.
(I hate just copying things but I also hate unwieldy YAML CI files... I don't expect us to change our branch naming convention often at all so this felt like possibly the least evil way to do it)